src/main/java/Duke.java

```diff
@@ -1,10 +1,128 @@
+
 public class Duke {
+
+    public static final String DIVIDER = "_________________________________________________";
+
+    public static void printIndividualTask(Task[] tasks, int taskNum) {
+        if (tasks[taskNum - 1].getTaskDescription().equals("todo")) {
+            System.out.println("Got it! You've added a todo task: ");
+            System.out.println(tasks[taskNum - 1]);
+            System.out.println(taskValidator(taskNum));
+        } else if (tasks[taskNum - 1].getTaskDescription().equals("deadline")) {
+            System.out.println("Got it! You've added a deadline task: ");
+            System.out.println(tasks[taskNum - 1]);
+            System.out.println(taskValidator(taskNum));
+        } else if (tasks[taskNum - 1].getTaskDescription().equals("event")) {
+            System.out.println("Got it! You've added an event task: ");
+            System.out.println(tasks[taskNum - 1]);
+            System.out.println(taskValidator(taskNum));
+        }
+    }
+
+    public static void printTaskList(Task[] tasks) {
+        System.out.println("Here are the tasks on your list: ");
+        for (int i = 1; i < tasks.length + 1; i++) {
+            if (tasks[i-1] == null) {
+                break;
+            } else {
+                String taskNum = Integer.toString(i);
+                System.out.println(taskNum + "." + tasks[i - 1]);
+            }
+        }
+    }
+
+    public static void printDoneTask(Task[] tasks, int taskNum) {
+        tasks[taskNum-1].markAsDone();
+        System.out.println("Awesome! I've marked the following task as done:");
+        System.out.println(tasks[taskNum-1]);
+    }
+
+
+    public static String taskValidator(int numTasks) {
+        String totalTasks = Integer.toString(numTasks);
+        if (numTasks <= 1) {
+            return "You now have " + totalTasks + " task in the list!";
+        } else {
+            return "You now have " + totalTasks+ " tasks in the list!";
+        }
+    }
+
+    public static void inputValidation(Task[] tasks, String userCommand) {
+        String[] words = userCommand.split(" ");
+        int spacesPadding = 1;
+        int taskPadding = 4;
+
+        String todoCommand = "todo";
+        String deadlineCommand = "deadline";
+        String eventCommand = "event";
+        String completeCommand = "done";
+        String listCommand = "list";
+
+        if (words[0].equals(todoCommand)) {
+            String todoTask = userCommand.substring(todoCommand.length() + spacesPadding);
+            Task todo = new Todo(todoTask);
+            tasks[todo.getTotalTasks()-1] = todo;
+            printIndividualTask(tasks,todo.getTotalTasks());
+        } else if (words[0].equals(deadlineCommand)) {
+            int indexOfBy = userCommand.indexOf("/by");
+            String deadlineTask = userCommand.substring(deadlineCommand.length() + spacesPadding, indexOfBy - 1);
+            String byDate = userCommand.substring(indexOfBy + taskPadding);
```

Comment:
Perhaps some of these variables are not necessary? For eg. it could just be userCommand.substring(9, indexOf("/")-1), or userCommand.substring(indexOfBy + 4); 
but if you added it for readability's sake then I guess that's fine too